zyte_common_items/processors.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, List, Optional
+
+from lxml.html import HtmlElement
+from parsel import Selector, SelectorList
+from web_poet.mixins import ResponseShortcutsMixin
+from zyte_parsers import Breadcrumb as zp_Breadcrumb
+from zyte_parsers import extract_breadcrumbs
+
+from .items import Breadcrumb
+
+
+def _get_base_url(page: Any) -> Optional[str]:
+    if isinstance(page, ResponseShortcutsMixin):
+        return page.base_url
+    return getattr(page, "url", None)
+
+
+def breadcrumbs_processor(value: Any, page: Any) -> Any:
+    def _from_zp_breadcrumb(value: zp_Breadcrumb) -> Breadcrumb:
+        return Breadcrumb(name=value.name, url=value.url)
+
+    if isinstance(value, SelectorList):
+        if len(value) == 0:
+            return None
+        value = value[0]
+
+    if isinstance(value, (Selector, HtmlElement)):
+        zp_breadcrumbs = extract_breadcrumbs(value, base_url=_get_base_url(page))
```

Comment:
Could you please add a test for base url handling?

Reply:
Do you want a test that uses `ResponseShortcutsMixin.base_url` to get one from HTML? As we already test the simplest case with `BasePage.url`.

Reply:
Yeah, make sure that if a page contains e.g. `<base>` tag, the right base url is used.